### PR TITLE
Someone needs to slow down!😉

### PR DESCRIPTION
**Pull request by Ryan-Myers**
Just pushing this as it has so many diffs, and variable type changes, it will impact m2c outputs in other places. So this will just make it easier.

### CODE REVIEW

File: src/lights.c [hunk 1]
```diff
@@ -60,18 +59,91 @@ void setup_lights(s32 count) {
...
     for (i = 0; i < gMaxLights; i++) {
         gActiveLights[i] = &D_800DC954[i];
     }
 }
-#else
-GLOBAL_ASM("asm/non_matchings/lights/setup_lights.s")
-#endif
 
+#ifdef NON_EQUIVALENT
+ObjectLight *func_80031CAC(Object* light, LevelObjectEntry_RgbaLight* lightEntry) {
```
Comment (FazanaJ) [inline src/lights.c:71]:
```suggestion
ObjectLight *func_80031CAC(Object *light, LevelObjectEntry_RgbaLight *lightEntry) {
```

File: src/lights.h [hunk 2]
```diff
@@ -94,7 +94,7 @@ void func_80032BAC(ObjectLight *light);
 s32 get_light_count(void);
 f32 light_distance_calc(ObjectLight *light);
 f32 light_direction_calc(ObjectLight *light);
-Object_64 *func_80031CAC(Object *, LevelObjectEntry_RgbaLight *entry);
+ObjectLight *func_80031CAC(Object* light, LevelObjectEntry_RgbaLight* lightEntry);
```
Comment (FazanaJ) [inline src/lights.h:97]:
```suggestion
ObjectLight *func_80031CAC(Object *light, LevelObjectEntry_RgbaLight *lightEntry);
```

File: src/lights.c [hunk 3]
```diff
@@ -60,18 +59,91 @@ void setup_lights(s32 count) {
     temp_v0 = (ObjectLight **) allocate_from_main_pool_safe(gMaxLights * (sizeof(s32 *) + sizeof(ObjectLight) + sizeof(unk800DC960) + sizeof(Vec3f)), COLOUR_TAG_MAGENTA);
     newCount = gMaxLights;
     gActiveLights = temp_v0;
-    D_800DC954 = (ObjectLight *) &temp_v0[newCount];
-    D_800DC960 = (unk800DC960 *) &D_800DC954[newCount];
-    D_800DC964 = (Vec3f *) &D_800DC960[newCount];
+    D_800DC954 = newCount + (0, temp_v0);
```
Comment (DavidSM64) [inline src/lights.c:62]: We should probably add a "Fakematch" comment here for the `(0, var)` statement.

**Review by DavidSM64 — APPROVED** [review]
LGTM

**Pull request merged**